### PR TITLE
Improve docstring for get_ground_diffuse

### PR DESCRIPTION
**Pull request by cwhanse**
- ~~[ ] Closes #xxxx~~
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~~[ ] Tests added~~
 - ~~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [X] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Reformat reference section and update links in `get_ground_diffuse`.

Fresh snow albedo value in `irradiance.SURFACE_ALBEDOS` conflicts with sources. Perhaps we change to 0.80 (wikipedia value); PVsyst uses 0.82.

### CODE REVIEW

File: pvlib/irradiance.py [hunk 1]
```diff
@@ -584,23 +579,25 @@ def get_ground_diffuse(surface_tilt, ghi, albedo=.25, surface_type=None):
...
-       http://files.pvsyst.com/help/albedo.htm
-       and
-       http://en.wikipedia.org/wiki/Albedo
-       and
-       https://doi.org/10.1175/1520-0469(1972)029<0959:AOTSS>2.0.CO;2
+    .. [2] https://www.pvsyst.com/help/albedo.htm Accessed January, 2024.
+    .. [3] http://en.wikipedia.org/wiki/Albedo Accessed January, 2024.
+    .. [4] Payne, R. E. "Albedo of the Sea Surface". J. Atmos. Sci., 29,
+       pp. 959–970, 1972.
+       https://doi.org/10.1175/1520-0469(1972)029<0959:AOTSS>2.0.CO;2.
```
Comment (echedey-ls) [inline pvlib/irradiance.py:600]: Possibly using the ``` :doi:` ... ` ``` role here is more appropriate and fixes the link problem?

**Comment by echedey-ls** [comment]
Handy link to new docs: https://pvlib-python--1953.org.readthedocs.build/en/1953/reference/generated/pvlib.irradiance.get_ground_diffuse.html

File: pvlib/irradiance.py [hunk 2]
```diff
@@ -584,23 +579,25 @@ def get_ground_diffuse(surface_tilt, ghi, albedo=.25, surface_type=None):
     Returns
     -------
     grounddiffuse : numeric
-        Ground reflected irradiance. [W/m^2]
+        Ground reflected irradiance. :math:`W/m^2`
 
+    Notes
+    -----
+    Ground diffuse irradiance is the the last term of equations 3, 4, 7, 8,
+    10, 11, and 12 in [1]_. Table of albedo values by ``surface_type``
```
Comment (adriesse) [inline pvlib/irradiance.py:587]: It might be more helpful to describe what that last term is, rather than refer to a paper that's not openly accessible.

Comment (cwhanse) [reply pvlib/irradiance.py]: There's a preprint [here](https://strathprints.strath.ac.uk/5008/1/Strachan_PA_et_al_Pure_Empirical_validation_of_models_to_compute_solar_irradiance_on_inclined_surfaces_for_building_energy_simulation_2007.pdf) but one has to find it by searching the title.

File: pvlib/irradiance.py [hunk 3]
```diff
@@ -551,13 +551,16 @@ def poa_components(aoi, dni, poa_sky_diffuse, poa_ground_diffuse):
...
+    Ground diffuse irradiance is calculated as
+
+    .. math::
+
+       G_{ground} = GHI \times \rho \times \frac{1 - \cos\beta}{2}
 
-    Function to determine the portion of irradiance on a tilted surface
-    due to ground reflections. Any of the inputs may be DataFrames or
-    scalars.
+    where :math:`\rho` is albedo and :math:`\beta` is ``surface_tilt``.
```
Comment (kandersolar) [inline pvlib/irradiance.py:563]:
```suggestion
    where :math:`\rho` is ``albedo`` and :math:`\beta` is ``surface_tilt``.
```

just for consistency

**Review by kandersolar — APPROVED** [review]

**Pull request merged**